rest-client-jdk/src/main/java/io/apicurio/rest/client/handler/RegistryDateDeserializer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Red Hat Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.rest.client.handler;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * A custom date deserializer for backwards compatibility.  Handles the date formats from 2.3.1.Final and also
+ * 2.4.0.Final (where we fixed the server-side date format returned by the API).
+ * @author eric.wittmann@gmail.com
+ */
+public class RegistryDateDeserializer extends JsonDeserializer<Date> {
+
+    private static final SimpleDateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```

Comment:
I think it would be best (for the future) to use the spec format first, hence quoting the Z and inverting the condition below.
But I like this solution! Thanks a lot for the simplification!

Reply:
I tested that and it didn't work for me.  When it was read in, the time was incorrect.  For example, if I used these inputs:

`2022-01-01T20:00:00Z`
and
`2022-01-01T20:00:00+0000`

When I read those both in using their respective formats, the resulting `Date` values were different.  The latter one was correct and the former was incorrect.

Likely I was doing something wrong (and I'm happy for someone to do it right), but the easy and obvious fix was to replace the Z with +0000 and then I got correct results.

Reply:
Oh, wow, ok, good enough for now, thanks for explaining 👍